server/src/main/java/io/druid/segment/loading/StorageLocationConfig.java

```diff
@@ -37,6 +37,10 @@
   @Min(1)
   private long maxSize = Long.MAX_VALUE;
 
+  // Negative value means that free space check is disabled.
+  @JsonProperty
+  private float freeSpacePercent = -1.0f;
```

Comment:
There's not a lot of places where `Float` would be appropriate with a `null` default but I think this can be one of them, where `0.0F` is a valid value.